ocaml/xenops/xenops.ml

```diff
@@ -228,7 +228,7 @@ let stat ~xs d =
 		| x -> string_of_kind x in
 	let frontend_type = match d.frontend.kind with
 		| Vbd _ | Tap ->
-			let be = frontend_path_of_device ~xs d in
+			let be = frontend_ro_path_of_device ~xs d in
 			(try if xs.Xs.read (sprintf "%s/device-type" be) = "cdrom" then "cdrom" else "disk" with _ -> "??")
```

Comment:
For this use here (with the inappropriately named `be` variable) we are only finding out whether an expression should return `"cdrom"`, `"disk"` or `"??"`, and for this we can use the information that came from the toolstack in the first place: there is no need to consult the guest software's claim about the type of device.